### ISSUE TITLE
add cross entropy loss function and ability to choose loss function when declaring self.SGD() 

### ISSUE THREAD

**Issue by a2tian** [WaiWasabi/Neural-Networks#2]

**Comment by a2tian** [comment]
done. Created the Cost class to be subclassed whenever a new cost is added. Each cost has two functions: fn(a, y) to calculate the cost of the network (mostly for analysis purposes, not actually used in gradient descent) and delta(z, a, y) to calculate the partial derivative of the cost function with respect to the weighted input, z,  in the output layer.

Added class variable "costs", a dictionary containing a string key that maps to the corresponding Cost subclass.

cost key is specified when declaring SGD
